test/e2e/suites/apivalidations/openstackmachine_test.go

```diff
@@ -63,6 +63,14 @@ var _ = Describe("OpenStackMachine API validations", func() {
 		By("Modifying the providerID")
 		machine.Spec.ProviderID = ptr.To("bar")
 		Expect(k8sClient.Update(ctx, machine)).NotTo(Succeed(), "Updating providerID should fail")
+
+		By("Setting the identityRef")
+		machine.Spec.IdentityRef = ptr.To(infrav1.OpenStackIdentityReference{Name: "foo", CloudName: "GCP"})
```

Comment:
Incidentally, it's not important to this test, except for readability, but `CloudName` here refers to an entry for an OpenStack cloud in `clouds.yaml`. So it's going to be something like `production`, `staging`, etc. In practise people don't tend to push a clouds.yaml with more than 1 cloud in it so this is academic, but we still support it.